packages/clay-badge/src/ClayBadge.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+import 'clay-icon';
+import templates from './ClayBadge.soy.js';
+import Component from 'metal-component';
+import Soy from 'metal-soy';
+import { Config } from 'metal-state';
+
+/**
+ * Metal Clay Badge component.
+ */
+class ClayBadge extends Component {}
+
+/**
+ * State definition.
+ * @static
+ * #type {!Object}
+ */
+ClayBadge.STATE = {
+	/**
+	 * Render ClayIcon in the ClayBadge element.
+	 * @instance
+	 * @memberof ClayBadge
+	 * @type {?Object|undefined}
+	 * @default undefined
+	 */
+	icon: Config.shapeOf({
+		alignment: Config.oneOf(['left', 'right']),
+		monospaced: Config.bool(),
+		spritemap: Config.string().required(),
+		symbol: Config.string().required(),
+	}),
+
+	/**
+	 * Id to be applied to the element.
+	 * @instance
+	 * @memberof ClayBadge
+	 * @type {?string|undefined}
+	 * default undefined
+	 */
+	id: Config.string(),
+
+	/**
+	 * The label of the badge element.
+	 * @instance
+	 * @memberof ClayBadge
+	 * @type {?string|html|undefined}
+	 * default undefined
+	 */
+	label: Config.any(),
+
+	/**
+	 * The size of the badge element.
+	 * @instance
+	 * @memberof ClayBadge
+	 * @type {?string|undefined}
+	 * default undefined
+	 */
+	size: Config.oneOf(['sm', 'lg']),
+
+	/**
+	 * Shape of the badge.
+	 * @instance
+	 * @memberof ClayBadge
+	 * @type {?string|undefined}
+	 * default undefined
+	 */
+	shape: Config.oneOf(['pill']),
```

Comment:
As I can see on Badges Clay docs there're no shapes for Badges, only that they can be pills or not. Maybe would be better to not use `shape` parameter for this but a boolean `pill` one.